projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -57,21 +57,15 @@ export class LinkValueComponent extends BaseValueComponent implements OnInit, On
    * @param searchTerm label to be searched
    */
   searchByLabel(searchTerm: string): ReadResource[] {
-    const linkType = this.parentResource.getLinkPropertyIriFromLinkValuePropertyIri(this.propType);
-    if (typeof linkType === 'string') {
-      this.restrictToResourceClass = this.parentResource.entityInfo.properties[linkType].objectType;
-    }
+    const linkType = this.parentResource.getLinkPropertyIriFromLinkValuePropertyIri(this.propIri);
+    this.restrictToResourceClass = this.parentResource.entityInfo.properties[linkType].objectType;
     // at least 3 characters are required
     if ( typeof searchTerm === 'string' && searchTerm.length >= 3) {
 
       this.knoraApiConnection.v2.search.doSearchByLabel(searchTerm, 0, {limitToResourceClass: this.restrictToResourceClass}).subscribe(
         (response: ReadResource[]) => {
           this.resources = response;
         });
-    } else {
-      // clear selection
-      this.resources = undefined;
```

Comment:
I think the idea of clearing the selection is good, what I meant is just `this.resources = []`